src/service/eol/eol.svc.ts

```diff
@@ -4,50 +4,63 @@ import type { Line } from '../line.ts';
 import type { ComponentStatus, ScanResult } from '../nes/modules/sbom.ts';
 import { NesApolloClient } from '../nes/nes.client.ts';
 import { createBomFromDir } from './cdx.svc.ts';
-import type { Sbom, SbomEntry, SbomMap as SbomModel, ScanOptions } from './eol.types.ts';
+import type { Sbom, ScanOptions } from './eol.types.ts';
 
-/**
- * Main function to scan directory and collect SBOM data
- */
-export async function scanForEol(directory = process.cwd(), opts: ScanOptions = {}) {
+export async function createSbom(directory: string, opts: ScanOptions = {}) {
   const sbom = await createBomFromDir(directory, opts.cdxgen || {});
   if (!sbom) throw new Error('SBOM not generated');
   log.info('SBOM generated');
-  const model = await extractComponents(sbom);
-  const scan = await submitScan(model);
-  // TODO: probably a better return option here...
-  return { model, sbom, scan };
+  return sbom;
+}
+
+export function validateIsCycloneDxSbom(sbom: unknown): asserts sbom is Sbom {
+  if (!sbom || typeof sbom !== 'object') {
+    throw new Error('SBOM must be an object');
+  }
+
+  const s = sbom as Record<string, unknown>;
+
+  // Basic CycloneDX validation
+  if (!('bomFormat' in s) || s.bomFormat !== 'CycloneDX') {
+    throw new Error('Invalid SBOM format: must be CycloneDX');
+  }
+
+  if (!('specVersion' in s) || typeof s.specVersion !== 'string') {
+    throw new Error('Invalid SBOM: missing specVersion');
+  }
+
+  if (!('components' in s) || !Array.isArray(s.components)) {
+    throw new Error('Invalid SBOM: missing or invalid components array');
+  }
+}
+
+/**
+ * Main function to scan directory and collect SBOM data
+ */
+export async function scanForEol(sbom: Sbom) {
+  const purls = await extractPurls(sbom);
+  const scan = await submitScan(purls);
+  return { purls, scan };
 }
 
 /**
- * Translate an SBOM to a SbomModel for easier handling.
+ * Translate an SBOM to a list of purls for api request.
  */
-export async function extractComponents(sbom: Sbom): Promise<SbomModel> {
-  const { components: comps, dependencies } = sbom;
-  const components = [...(comps || []), ...(dependencies || [])].reduce(
-    (acc, entry) => {
-      if (entry.purl) {
-        acc[entry.purl] = entry;
-      }
-
-      return acc;
-    },
-    {} as Record<string, SbomEntry>,
-  );
-
-  return { components, purls: Object.keys(components).sort() };
+export async function extractPurls(sbom: Sbom): Promise<string[]> {
+  const { components: comps } = sbom;
+  return comps?.map((c) => c.purl) ?? [];
```

Comment:
```suggestion
  return comps.map((c) => c.purl);
```

Technically we could trust that the input is always valid since its validated by `validateIsCycloneDxSbom `